assertj-error-prone/src/test/java/com/palantir/assertj/errorprone/AssertjCollectionAssertTest.java

```diff
@@ -0,0 +1,138 @@
+                        "import com.google.common.collect.ImmutableList;",
+                        "import java.util.ArrayList;",
+                        "import java.util.List;",
+                        "public class Test {",
+                        "  void f(List<String> actual, List<String> expected) {",
+                        "    assertThat(actual).containsExactlyElementsOf(expected);",
+                        "    assertThat(actual).containsExactlyElementsOf(new ArrayList<>());",
+                        "    assertThat(actual).isEmpty();",
+                        "    assertThat(actual).isEmpty();",
+                        "    assertThat(actual).containsExactly(\"foo\", \"bar\");",
+                        "    assertThat(actual).containsExactly(\"foo\", \"bar\");",
+                        "  }",
+                        "}")
+                .doTest(BugCheckerRefactoringTestHelper.TestMode.TEXT_MATCH);
+    }
+
+    @Test
+    public void fix_set() {
+        test().addInputLines(
+                        "Test.java",
+                        "import static org.assertj.core.api.Assertions.assertThat;",
+                        "import com.google.common.collect.ImmutableSet;",
+                        "import java.util.HashSet;",
+                        "import java.util.Set;",
+                        "public class Test {",
+                        "  void f(Set<String> actual, Set<String> expected) {",
+                        "    assertThat(actual).isEqualTo(expected);",
+                        "    assertThat(actual).isEqualTo(new HashSet<>());",
+                        "    assertThat(actual).isEqualTo(Set.of());",
+                        "    assertThat(actual).isEqualTo(ImmutableSet.of());",
+                        "    assertThat(actual).isEqualTo(Set.of(\"foo\", \"bar\"));",
+                        "    assertThat(actual).isEqualTo(ImmutableSet.of(\"foo\", \"bar\"));",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import static org.assertj.core.api.Assertions.assertThat;",
+                        "import com.google.common.collect.ImmutableSet;",
+                        "import java.util.HashSet;",
+                        "import java.util.Set;",
+                        "public class Test {",
+                        "  void f(Set<String> actual, Set<String> expected) {",
+                        "    assertThat(actual).containsExactlyInAnyOrderElementsOf(expected);",
+                        "    assertThat(actual).containsExactlyInAnyOrderElementsOf(new HashSet<>());",
+                        "    assertThat(actual).isEmpty();",
+                        "    assertThat(actual).isEmpty();",
+                        "    assertThat(actual).containsExactlyInAnyOrder(\"foo\", \"bar\");",
+                        "    assertThat(actual).containsExactlyInAnyOrder(\"foo\", \"bar\");",
```

Comment:
`ImmutableSet` guarantees order, however order is not taken into account in the equality check iirc. It's certainly an odd case